LanguageFeatures/Extension-types/static_analysis_extension_types_A17_t04.dart

```diff
@@ -2,33 +2,27 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
-/// @assertion Let DV be an extension type declaration named Name with type
-/// parameters X1 extends B1, .. Xs extends Bs. Assume that the representation
-/// declaration of DV is (R id)
+/// @assertion An extension type `V` is a proper subtype of `Object?`. It is
+/// potentially non-nullable, unless it implements `Object` or a subtype thereof
 ///
-/// We then say that the declared representation type of Name is R, and the
-/// instantiated representation type corresponding to Name<T1,.. Ts> is
-/// [T1/X1, .. Ts/Xs]R.
-/// ...
-/// Let V be an extension type of the form Name<T1, .. Ts>, and let R be the
-/// corresponding instantiated representation type. If R is non-nullable then V
-/// is a proper subtype of Object, and V is non-nullable. Otherwise, V is a
-/// proper subtype of Object?, and V is potentially nullable.
-///
-/// @description Checks that if an instantiated representation type `R` is
-/// not non-nullable then an extension type `V` is potentially nullable (it's a
-/// compile-time error to assign `Object?` or `null` to `V`)
+/// @description Checks that if an extension type doesn't implement a subtype of
+/// `Object` then is potentially non-nullable (it's a compile-time error to
+/// assign `Object?` or `null` to it)
```

Comment:
The idea here is to test that an extension type is a proper subtype of `Object?`, i.e. it can be assigned to `Object?` but `Object?` cannot be assigned to it. And this test checks the latter. Do you believe that we should check here assigning of the `null` only?

Reply:
Looking again, it seems likely that the test where 'proper subtype' should actually be tested is LanguageFeatures/Extension-types/static_analysis_extension_types_A17_t03.dart.

So if the tests about assigning `Object?` to an extension typed variable are added over there and deleted from here then I think we'll have a good alignment between the contents of the tests and the descriptions of the tests.

Of course there could be other ways to reorganize that would create a similarly precise alignment, but I do think it would be great if something is done about the fact that t03 says "proper subtype" but doesn't test more than half of it, and this test doesn't say "proper subtype", but it _does_ test the halfpart that was missing from t03. ;-)